pkgs/development/libraries/hipcub/default.nix

```diff
@@ -1,25 +1,44 @@
-{stdenv, fetchFromGitHub, cmake, rocm-cmake, pkgconfig, hip, rocprim}:
+{stdenv, fetchFromGitHub, lib, config, cmake, pkgconfig
+, rocm-cmake, hip, rocprim
+, doCheck ? false, gtest ? null
+}:
+
+assert doCheck -> gtest != null;
+
 stdenv.mkDerivation rec {
   name = "hipcub";
-  version = "3.5.0";
+  version = "3.8.0";
   src = fetchFromGitHub {
     owner = "ROCmSoftwarePlatform";
     repo = "hipCUB";
     rev = "rocm-${version}";
-    sha256 = "186kbwpy0kq5kg9cxsa9p170rdn1cz3sar69nlrqjzyx5alx1i9k";
+    sha256 = "0l1a0lgwzad2gfpjvmix6372dksm6vjivsc9qk4lyjy6dhdlsljb";
   };
-  patchPhase = ''
-    sed -e '/find_package(Git/,/endif()/d' \
-        -e '/download_project(/,/^[[:space:]]*)/d' \
-        -i cmake/Dependencies.cmake
-    sed 's,include(cmake/VerifyCompiler.cmake),,' -i CMakeLists.txt
-  '';
+
+  inherit doCheck;
+  
   nativeBuildInputs = [ cmake rocm-cmake pkgconfig ];
+  
+  buildInputs = [ hip rocprim ];
+
+  checkInputs = [ gtest ];
+
   cmakeFlags = [
     "-DCMAKE_CXX_COMPILER=hipcc"
-    # "-DHIP_PLATFORM=clang"
     "-DCMAKE_INSTALL_INCLUDEDIR=include"
-    "-DBUILD_TEST=NO"
+    "-DAMDGPU_TARGETS=${lib.strings.concatStringsSep ";" (config.rocmTargets or ["gfx803" "gfx900" "gfx906"])}"
+    "-DBUILD_TEST=${if doCheck then "YES" else "NO"}"
+    #"${if doCheck then "-DAMDGPU_TEST_TARGETS=${lib.strings.concatStringsSep ";" (config.rocmTargets or ["gfx803" "gfx900" "gfx906"])}" else ""}"
   ];
-  buildInputs = [ hip rocprim ];
-}
+
+  patchPhase = ''
+    sed -e '/find_package(Git/,/endif()/d' \
+        -e '/download_project(/,/^[[:space:]]*)/d' \
+        -i cmake/Dependencies.cmake
+    #sed 's,include(cmake/VerifyCompiler.cmake),,' -i CMakeLists.txt
```

Comment:
Can you delete this line that removes the `VerifyCompiler` inclusion? No need to keep it commented out as it should live on in git history.

Reply:
This should be fixed now, but I ended up including it in the `hipcub` commit.